lib/build_ir.cpp

```diff
@@ -0,0 +1,132 @@
+/* 
+ * Kehu - a toe language
+ * Copyright (C) 2023 Zhen You Zhe
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include <kehu/build_ir.h>
+#include <llvm/IR/IRBuilder.h>
+
+namespace kehu::ir
+{
+
+using llvm::IRBuilder;
+using llvm::Value;
+using llvm::LLVMContext;
+using llvm::Module;
+
+static Value *build_raw_number_ir(
+                std::shared_ptr<const ast::raw_integer_node> node,
+                IRBuilder<> &builder, LLVMContext &context)
+{
+        llvm::ConstantFP *number = llvm::ConstantFP::get(context,
+                llvm::APFloat(static_cast<float>(node->value)));
+        return number;
+}
+
+static llvm::GlobalValue *build_variable_definition_ir(
+                std::shared_ptr<const ast::definition_node> node,
+                IRBuilder<> &builder, LLVMContext &context, Module &module)
+{
+        llvm::GlobalVariable *var = nullptr;
+        return var;
+}
+
+/**
+ * @brief Get the function type object
+ * 
+ * Make the function type:  double(double,double) etc.
+ */
+static llvm::FunctionType *get_function_type(
+                std::shared_ptr<const ast::function_definition_node> fdef,
```

Comment:
There are too much parameters. Should use a class to collect them all? Or even OOP?